portability-spi-gateway/src/main/java/org/dataportabilityproject/spi/gateway/auth/AuthServiceProviderRegistry.java

```diff
@@ -4,11 +4,25 @@
  * Manages {@link AuthServiceProvider}s registered in the system.
  */
 public interface AuthServiceProviderRegistry {
+  /**
+   * Lookup the @code{AuthDataGenerator} corresponding to the serviceId, transferType and AuthMode
+   * specified
+   *
+   * @param serviceId the AuthServiceProvider to use
+   * @param transferDataType The TransferdataType to authorize for
+   * @param mode The authorization mode
+   * @return An AuthDataGenerator from the specified AuthServiceProvider for the type and mode
+   * requested.
+   */
+  AuthDataGenerator getAuthDataGenerator(String serviceId, String transferDataType, AuthMode mode);
 
-    /**
-     * Returns the provider that supports the service id.
-     *
-     * @param serviceId the service id
-     */
-    AuthServiceProvider getServiceProvider(String serviceId);
+  /**
+   * The AuthorizationMode to use for lookups.
+   * IMPORT specifies an authorization that allows you to import a type (implying read-write permissions)
+   * EXPORT specifies an authorization that allows you to export a type (implying read-only permissions)
+   */
+  enum AuthMode {
```

Comment:
@holachuy  I know you added TransferMode in #134 - Does it make sense to either switch to that, or move that in here before merging?

Reply:
I don't mind going with AuthMode but the bigger question seems that it goes unused except in log statements. Can we remove it alltogether?

Reply:
I was thinking it will get passed into the getAuthDataGenerator call to the AuthServiceProvider, which will then return a generator for the correct permissions mode.

Reply:
Per offline discussion, SGTM

Reply:
Done - Added test. Also for posterity, this is meant to mimic the existing ServiceMode enum here: https://github.com/google/data-portability/blob/master/portability-core/src/main/java/org/dataportabilityproject/shared/ServiceMode.java

Used to determine scopes for auth generation (see GoogleServiceProvider for example: https://github.com/google/data-portability/blob/master/portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/GoogleServiceProvider.java#L64)